core/src/main/java/org/elasticsearch/indices/store/IndicesStore.java

```diff
@@ -146,27 +185,12 @@ boolean shardCanBeDeleted(ClusterState state, IndexShardRoutingTable indexShardR
 
         for (ShardRouting shardRouting : indexShardRoutingTable) {
             // be conservative here, check on started, not even active
-            if (!shardRouting.started()) {
+            if (shardRouting.started() == false) {
                 return false;
             }
 
-            // if the allocated or relocation node id doesn't exists in the cluster state  it may be a stale node,
-            // make sure we don't do anything with this until the routing table has properly been rerouted to reflect
-            // the fact that the node does not exists
-            DiscoveryNode node = state.nodes().get(shardRouting.currentNodeId());
```

Comment:
I don't think we have any assertion for this. The tricky thing is finding a good place for it (might be a bit too expensive to check it whenever a new cluster state is built).